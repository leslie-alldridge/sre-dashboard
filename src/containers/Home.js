import React, { Component } from "react";
import {
  HeaderBar,
  LinkButtons,
  loginButton,
  registerButton
} from "../components";

const title = {
  pageTitle: "Home Screen"
};

class Home extends Component {
  render() {
    return (
      <div className="home-page">
        <HeaderBar title={title} />
        <LinkButtons
          buttonText={`Register`}
          buttonStyle={registerButton}
          link={`/register`}
        />
        <LinkButtons
          buttonText={`Login`}
          buttonStyle={loginButton}
          link={`/login`}
        />
        <LinkButtons
          buttonText={`Dash`}
          buttonStyle={loginButton}
          link={`/dashboard`}
        />
      </div>
    );
  }
}

export default Home;
