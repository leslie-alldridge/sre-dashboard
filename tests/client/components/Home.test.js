import React from "react";
import { shallow } from "enzyme";
import Home from "../../../src/containers/Home";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Home debug />);

    expect(component).toMatchSnapshot();
  });
});
