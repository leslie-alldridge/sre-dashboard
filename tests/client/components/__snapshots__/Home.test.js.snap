// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Home Screen",
            }
          }
        />,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "green",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Register"
          link="/register"
        />,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Login"
          link="/login"
        />,
      ],
      "className": "home-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Home Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "green",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Register",
          "link": "/register",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "royalblue",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Login",
          "link": "/login",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Home Screen",
              }
            }
          />,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "green",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Register"
            link="/register"
          />,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "royalblue",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Login"
            link="/login"
          />,
        ],
        "className": "home-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Home Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "green",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Register",
            "link": "/register",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "royalblue",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Login",
            "link": "/login",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
